examples/mdetr/data/transforms.py

```diff
@@ -0,0 +1,459 @@
+        return F.to_tensor(img), target
+
+
+class Normalize(object):
+    def __init__(self, mean, std):
+        self.mean = mean
+        self.std = std
+
+    def __call__(self, image, target=None):
+        image = F.normalize(image, mean=self.mean, std=self.std)
+        if target is None:
+            return image, None
+        target = target.copy()
+        h, w = image.shape[-2:]
+        if "boxes" in target:
+            boxes = target["boxes"]
+            boxes = box_convert(boxes, "xyxy", "cxcywh")
+            boxes = boxes / torch.tensor([w, h, w, h], dtype=torch.float32)
+            target["boxes"] = boxes
+        return image, target
+
+
+class Compose(object):
+    def __init__(self, transforms):
+        self.transforms = transforms
+
+    def __call__(self, image, target):
+        for t in self.transforms:
+            image, target = t(image, target)
+        return image, target
+
+    def __repr__(self):
+        format_string = self.__class__.__name__ + "("
+        for t in self.transforms:
+            format_string += "\n"
+            format_string += "    {0}".format(t)
+        format_string += "\n)"
+        return format_string
+
+
+class MDETRTransform:
+    IMAGENET_MEAN = [0.485, 0.456, 0.406]
+    IMAGENET_STD = [0.229, 0.224, 0.225]
+
+    def __init__(self, tokenizer: Callable, is_train: bool, cautious: bool = True):
+        normalize = Compose(
+            [ToTensor(), Normalize(self.IMAGENET_MEAN, self.IMAGENET_STD)]
+        )
+        self.tokenizer = tokenizer
+        scales = [480, 512, 544, 576, 608, 640, 672, 704, 736, 768, 800]
+        max_size = 1333
+        if is_train:
+            horizontal = [] if cautious else [RandomHorizontalFlip()]
+            self.image_transform = Compose(
+                horizontal
+                + [
+                    RandomSelect(
+                        RandomResize(scales, max_size=max_size),
+                        Compose(
+                            [
+                                RandomResize([400, 500, 600]),
+                                RandomSizeCrop(384, max_size, respect_boxes=cautious),
+                                RandomResize(scales, max_size=max_size),
+                            ]
+                        ),
+                    ),
+                    normalize,
+                ]
+            )
+
+        else:
+            self.image_transform = Compose(
+                [
+                    RandomResize([800], max_size=max_size),
+                    normalize,
+                ]
+            )
+
+    def __call__(
+        self, image: Union[Iterable[Image], Image], target: Dict[str, Any]
+    ) -> torch.Tensor:
+        image, target = self.image_transform(image, target)
+        target["tokenized"] = self.tokenizer(target["caption"], return_tensors="pt")[
+            "input_ids"
+        ][0]
+        return image, target
+
+
+def create_positive_map(tokenized, tokens_positive):
```

Comment:
can you just use the `construct_positive_map` from your contrastive alignment loss?

Reply:
Not as things are currently written. That one relies on the indices output by the matcher to order the positive map for the loss calculation. I have a to-do to try and refactor to a common implementation, but it is low-pri at the moment relative to other things.